stdlib/public/runtime/Once.cpp

```diff
@@ -41,6 +48,18 @@ void swift::swift_once(swift_once_t *predicate, void (*fn)(void *)) {
 #if defined(__APPLE__)
   dispatch_once_f(predicate, nullptr, fn);
 #else
+#if defined(__CYGWIN__)
+
+  mutex_.lock();
```

Comment:
Please add a fixme (and file a bug) that notes that this implementation does a global lock, which is much worse than what we have on other platforms.  Each `swift_once` should synchronize on the token.